counsel.el

```diff
@@ -191,11 +191,9 @@ Update the minibuffer with the amount of lines collected every
       (with-current-buffer (process-buffer process)
         (goto-char (point-min))
         (setq size (- (buffer-size) (forward-line (buffer-size))))
-        (ivy--set-candidates
-         (split-string
-          (buffer-string)
-          counsel-async-split-string-re
-          t)))
+        (let ((cands (split-string (buffer-string) counsel-async-split-string-re t)))
+          (setq ivy--old-cands ivy--all-candidates)
```

Comment:
If there's another message to be handled that you can reproduce, go ahead and add it. But this stuff has worked for a very long time with no complaints.